webrtc.html

```diff
@@ -11090,19 +11090,73 @@ <h3 id="mediastreamtrack-network-use">MediaStreamTrack</h3>
+        sourced by <code>getUserMedia()</code>. Only constraints explicitly
+        specified in the context of WebRTC are applicable.
+        The settings of a track sourced from an <code>RTCPeerConnection</code>
+        represent the currently received frames/samples after downsampling based
+        on applied constraints, if applied.
+        <code>MediaStreamTrack.getCapabilities()</code> MUST always return the
+        empty set.</p>
+        <p>The following constrainable properties are defined to apply only to
+        remote video <code><a>MediaStreamTrack</a></code> objects generated by
+        an <code>RTCPeerConnection</code>:</p>
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>Property Name</th>
+              <th>Values</th>
+              <th>Notes</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr id="def-constraint-width">
+              <td><dfn>width</dfn></td>
+              <td><code><a>ConstrainULong</a></code></td>
+              <td>The width or width range, in pixels. The constrainable
+              property allows downscaling. As a setting, this is the width of
+              the latest frame.</td>
+            </tr>
+            <tr id="def-constraint-height">
+              <td><dfn>height</dfn></td>
+              <td><code><a>ConstrainULong</a></code></td>
+              <td>The height or height range, in pixels. The constrainable
+              property allows downscaling. As a setting, this is the height of
+              the latest frame.</td>
+            </tr>
+            <tr id="def-constraint-frameRate">
+              <td><dfn>frameRate</dfn></td>
+              <td><code><a>ConstrainDouble</a></code></td>
+              <td>The exact frame rate (frames per second) or frame rate range.
+              The constrainable property allows downsampling by dropping frames.
+              As a setting, this is an estimate of the frame rate based on
+              recently received frames.</td>
+            </tr>
+            <tr id="def-constraint-aspect">
+              <td><dfn>aspectRatio</dfn></td>
+              <td><code><a>ConstrainDouble</a></code></td>
+              <td>The exact aspect ratio (width in pixels divided by height in
+              pixels, represented as a double rounded to the tenth decimal
+              place) or aspect ratio range. The constrainable property allows
+              downscaling. As a setting, this is the aspect ratio of the latest
+              frame.</td>
+            </tr>
+            <tr id="def-constraint-resizeMode">
+              <td><dfn>resizeMode</dfn></td>
+              <td><code><a>ConstrainDOMString</a></code></td>
+              <td>Downscaling can happen if constraints are applied to width or
+              height. The resize mode affects how the User Agent downscales
+              frames, as defined in [[!GETUSERMEDIA]].</td>
+            </tr>
+          </tbody>
+        </table>
```

Comment:
I think we need a chapter on *how* downscaling and frame decimation MUST be done, and how the *applyConstraints* is to be implemented, like we do in https://w3c.github.io/mediacapture-screen-share/#downscaling-and-frame-decimation

Reply:
Can I simply reference that section?

Reply:
Editors said they don't want webrtc-pc to normatively reference screen-capture.

Also, screen-capture disallows cropping. Doesn't Chrome support cropping? If we want that here we should perhaps provide our own section.

Sadly, there may not be one downscaling algorithm to rule them all. (Yes I know the remote track may be a screen sharing track, but we can't have it all I guess).

Reply:
Taking another look, resizeMode is underspecified in gUM (I filed https://github.com/w3c/mediacapture-main/issues/584). In terms of remote track, "none" doesn't make any sense. I think resizeMode shouldn't be applicable to remote tracks. But we need to do make sure it is specified how to downscale it. I'll check with @guidou what Chrome does.